### PR TITLE
Added script length checking

### PR DESCRIPTION
**Pull request by Seggan**
## Description
I added a script length check right before opening the edit script page. The reason for this is that in an addon I'm making, I can set much longer scripts than 54 instructions.

## Changes
Added a script length check

## Related Issues
N/A

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos. (I don't think this needs testing)
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java [hunk 1]
```diff
@@ -470,7 +470,12 @@ public void openScriptEditor(Player p, Block b) {
 
         menu.addItem(1, new CustomItem(HeadTexture.SCRIPT_FORWARD.getAsItemStack(), "&2> Edit Script", "", "&aEdits your current Script"));
         menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
-            openScript(pl, b, getScript(b.getLocation()));
+            if (PatternUtils.DASH.split(BlockStorage.getLocationInfo(b.getLocation()).getString("script")).length <= MAX_SCRIPT_LENGTH) {
+                openScript(pl, b, getScript(b.getLocation()));
+            } else {
+                pl.closeInventory();
+                pl.sendMessage(ChatColor.RED + "The script is too large to edit!");
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:477]: Message isn't localized

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java [hunk 2]
```diff
@@ -470,7 +470,12 @@ public void openScriptEditor(Player p, Block b) {
 
         menu.addItem(1, new CustomItem(HeadTexture.SCRIPT_FORWARD.getAsItemStack(), "&2> Edit Script", "", "&aEdits your current Script"));
         menu.addMenuClickHandler(1, (pl, slot, item, action) -> {
-            openScript(pl, b, getScript(b.getLocation()));
+            if (PatternUtils.DASH.split(BlockStorage.getLocationInfo(b.getLocation()).getString("script")).length <= MAX_SCRIPT_LENGTH) {
+                openScript(pl, b, getScript(b.getLocation()));
+            } else {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:475]: This check is already performed in #setScript(...).
So no real need for that.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java (hunk 2 above)
Comment (Seggan) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:475]: That's the entire reason I made the pr. I have the need for scripts of 100 instructions or more. setscript won't allow me to do that but I can use blockstorage to circumvent that. However if a player edits a script of such a big size they will get an inventory error. That's why I added the second check

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:475]: :NotLikeThis:

Comment (Seggan) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:475]: So...

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java [hunk 3]
```diff
@@ -551,7 +556,6 @@ public void setScript(@Nonnull Location l, @Nonnull String script) {
         Validate.notNull(script, "No script given");
         Validate.isTrue(script.startsWith(Instruction.START.name() + '-'), "A script must begin with a 'START' token.");
         Validate.isTrue(script.endsWith('-' + Instruction.REPEAT.name()), "A script must end with a 'REPEAT' token.");
-        Validate.isTrue(PatternUtils.DASH.split(script).length <= MAX_SCRIPT_LENGTH, "Scripts may not have more than " + MAX_SCRIPT_LENGTH + " segments");
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/ProgrammableAndroid.java:554]: Dear lord. Do not remove this :NotLikeThis:

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]
The reason as to *why* this was added, I don't agree so much with.
But error-prevention and input validation is always good to have, so... approved

**Review by WalshyDev — APPROVED** [review]

**Pull request merged**